packages/web-components/fast-components/src/design-system-provider/README.md

```diff
@@ -0,0 +1,251 @@
+import { cssCustomPropertyBehaviorFactory, DesignSystemProvider } from "@microsoft/fast-components";
+
+const styles = css`
+    :host {
+        height: var(--fancy-height);
+    }
+`.withBehaviors(
+    cssCustomPropertyBehaviorFactory(
+        "fancy-height",
+        designSystem => Math.pow(designSystem.designUnit, 12),
+        DesignSystemProvider.findProvider
+    )
+)
+```
+
+### Creating a Design System Provider
+
+To create a new _Design System Provider_, extend the `DesignSystemProvider` class and decorate it with the `@designSystemProvider` decorator, providing the decorator the element _tag name_ you wish to use:
+
+**EXAMPLE: Creating a custom Design System Provider**
+
+```ts
+import { DesignSystemProvider, designSystemProvider } from "@microsoft/fast-components";
+
+@designSystemProvider("fancy-design-system-provider")
+class FancyDesignSystemProvider extends DesignSystemProvider {}
+```
+
+```html
+<fancy-design-system-provider use-defaults>
+    Hello Fancy World!
+</fancy-design-system-provider>
+```
+
+#### Declaring Design System Properties
+
+Building off the above, _design system_ properties can be declared using the `@designSystemProperty` decorator.
+
+**EXAMPLE: Creating a design system property declaration**
+
+```ts
+// ..
+@designSystemProvider("fancy-design-system-provider")
+class FancyDesignSystemProvider extends DesignSystemProvider {
+    @attr("fancy-property")
+    @designSystemProperty({ cssCustomProperty: "fancy-property", default: "red" })
+    public fancyProperty;
+}
+```
+
+```html
+<fancy-design-system-provider use-defaults>
+    <p style="color: var(--fancy-property)">Red text!</p>
+</fancy-design-system-provider>
+```
+
+Declaring a _design system_ property will do several things. By default, the property name being decorated will become a CSS custom property when the value is set. This name can be customized by setting `cssCustomProperty` on the provided config object.
+
+```ts
+// Will create a CSS custom property accessible by `var(--fooBar)`
+@designSystemProperty({default: "foo"})
+public fooBar;
+```
+
+```ts
+// Will create a CSS custom property accessible by `var(--foo-bar)`
+@designSystemProperty({default: "foo", cssCustomProperty: "foo-bar"})
+public fooBar;
+```
+
+Additionally, creation of CSS custom properties can also be disabled by setting `cssCustomProperty` to `false`;
+
+```ts
+// No CSS custom property will be created
+@designSystemProperty({default: "foo", cssCustomProperty: false})
+public fooBar;
+```
+
+The default value _must_ be set through the `default` property on the supplied config. This is the value that will be applied when the `use-defaults` attribute is used and no override property is provided.
+
+### FAST Design System Properties
+|Property Name|Type|Attribute Name|CSS Custom property|
+|---|---|---|---|
+|backgroundColor|string| background-color,  | background-color |
+|accentBaseColor|string| accent-base-color | N/A |
```

Comment:
Should the `accentBaseColor` be provided as a CSS property?

Reply:
It generally should not be used directly and doesn't *currently* exist as a CSS custom property.

The reason for that is because the *base color* derives the accent palette, and UI colors are derived using recipes *from* that palette. There are cases where the color resolved by the recipe may match the `accentBaseColor` but also cases where it would not - resulting in contrast failures.